src/clients/kol.ts

```diff
@@ -362,22 +362,8 @@ export class KoLClient {
   async getPartialPlayerFromId(id: number): Promise<PartialPlayer | null> {
     try {
       const profile = await this.tryRequestWithLogin("showplayer.php", { who: id });
-      const header = profile.match(/<b>([^>]*?)<\/b> \(#(\d+)\)<br>/);
-      if (!header) return null;
-
-      const goldstars =
-        profile.match(/<img src="\/images\/otherimages\/goldstart.png" height="30" width="30" \/>/g)
-          ?.length ?? 0;
-      const optimalityMultiplier = 10 ** (goldstars / 2);
-      const level = Number(header[2] || "0") / optimalityMultiplier;
-      const clss = profile.match(/<b>Class:<\/b><\/td><td>(.*?)<\/td>/)?.[1] ?? "Unknown";
-
-      return {
-        id: id,
-        name: header[1],
-        level,
-        class: clss,
-      };
+      const name = profile.match(/<b>([^>]*?)<\/b> \(#(\d+)\)<br>/)?.[1];
+      return name ? await this.getPartialPlayerFromName(name) : null;
```

Comment:
the one small thing in here is that i am sad that we removed the goldstars lookup even though we very obviously were not using it or exposing it anywhere. but i am ok with this even if it's 2 pings for id lookups vs 1 ping for name lookups